src/main/java/cubicchunks/server/chunkio/async/forge/AsyncWorldIOExecutor.java

```diff
@@ -73,22 +72,27 @@
 	 *
 	 * @param world The world in which the cube lies
 	 * @param loader The file loader for cubes
-	 * @param y cube y position
-	 * @param column The column of the cube
+	 * @param cache the cube cache used to load cubes and columns
+	 * @param cubeX X coordinate of the cube to load
+	 * @param cubeY Y coordinate of the cube to load
+	 * @param cubeZ Z coordinate of the cube to load
 	 *
 	 * @return The loaded cube, or null if either not present or the load failed
 	 */
 	@Nullable
-	public static Cube syncCubeLoad(World world, CubeIO loader, int y, Column column) {
-		QueuedCube key = new QueuedCube(column.getX(), y, column.getZ(), world);
+	public static Cube syncCubeLoad(ICubicWorld world, CubeIO loader, ServerCubeCache cache, int cubeX, int cubeY, int cubeZ) {
+		Column column = cache.loadChunk(cubeX, cubeZ);
+		QueuedCube key = new QueuedCube(cubeX, cubeY, cubeZ, world);
 		AsyncCubeIOProvider task = cubeTasks.remove(key); // Remove task because we will call the sync callbacks directly
 		if (task != null) {
 			runTask(task);
 		} else {
-			task = new AsyncCubeIOProvider(key, column, loader);
+			task = new AsyncCubeIOProvider(key, loader);
+			task.setColumn(column);
 			task.run();
+			task.runSynchronousPart();
 		}
-		task.runSynchronousPart();
+		task.runCallbacks();
```

Comment:
This will run callbacks twice - you didn't remove the runCallbacks() part in the cube provider